library/ssl_cli.c

```diff
@@ -642,7 +637,8 @@ static void ssl_write_alpn_ext( mbedtls_ssl_context *ssl,
     /* Skip writing extension and list length for now */
     p += 4;
 
-    for( cur = ssl->conf->alpn_list; *cur != NULL; cur++ )
+    for( cur = ssl->conf->alpn_list;
+	 p < buf + 6 + alpnlen && cur != NULL; cur++ )
```

Comment:
No it's not, because the `cur` list might be too long, in which case `alpnlen` still provides a valid bound. The alternative would be to fail if `cur` is too large.